src/WebServer_JSON.ino

```diff
@@ -30,9 +31,17 @@ void handle_rawval()
     if (validDeviceIndex(DeviceIndex))
     {
       LoadTaskSettings(taskNr);
-      htmlData += String(ExtraTaskSettings.TaskDeviceValueNames[valNr]);
-      htmlData += " ";
-      htmlData += formatUserVarNoCheck(taskNr, valNr);
+
+      for (byte x = 0; x < Device[DeviceIndex].ValueCount; x++)
+      {
+        if (valNr == INVALID_TASKVAR_INDEX || valNr == x)
+        {
+          htmlData += String(ExtraTaskSettings.TaskDeviceValueNames[x]);
```

Comment:
Please try to make some estimate on the expected size of `htmlData` and call `htmlData.reserve()` before appending.
For example use an estimate of 16 bytes per var name and 6 per value + ';' + '\n' = 24 bytes per value.
So a rough estimate can be to reserve 128 bytes..